### PR TITLE
fix(content-item): add text overflow ellipsis to avoid overlap, add p…

### PR DESCRIPTION
**Pull request by daanishnasir**
…osition relative and box-shadow

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/collab-ui/collab-ui-core/blob/master/CONTRIBUTING.md#commit-message-format
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")
- [x] Bugfix
- [ ] Feature
- [ ] Documentation update
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

add ellipsis overflow to avoid overlap, fix for actionNode buttons on IconView, was moving. Border changed to box-shadow. 

**Does this PR introduce a breaking change?** (check one with "x")
- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: scss/components/content-item.scss [hunk 1]
```diff
@@ -51,6 +51,7 @@
       margin-bottom: 8px;
       background-color: $gray-7;
       border-radius: 4px;
+      box-shadow: 0 0 0 1px rgba(0, 0, 0, 0.2);
```
Comment (bfbiggs) [inline scss/components/content-item.scss]: Should use the variable here.

File: scss/components/content-item.scss [hunk 2]
```diff
@@ -67,6 +68,7 @@
 
     &--failed {
       background-color: $gray-7;
+      box-shadow: 0 0 0 1px rgba(0, 0, 0, 0.2);
```
Comment (pauljeter) [inline scss/components/content-item.scss]: same as above

File: scss/components/content-item.scss [hunk 3]
```diff
@@ -140,11 +142,15 @@
         flex-direction: column;
 
         &--file-name {
+          width: 14rem;
```
Comment (pauljeter) [inline scss/components/content-item.scss]: Use rem-calc with pixels incase REMs change in the system

File: scss/components/content-item.scss [hunk 4]
```diff
@@ -140,11 +142,15 @@
...
           height: 24px;
+          overflow: hidden;
```
Comment (pauljeter) [inline scss/components/content-item.scss]: Do we have an ellipsis mixin that adds these 2 properties?

File: scss/components/content-item.scss [hunk 5]
```diff
@@ -304,16 +314,19 @@
...
       font-size: 14px;
       line-height: 22px;
       color: $gray-2;
       text-align: center;
+      text-overflow: ellipsis;
+      white-space: nowrap;
     }
   }
 
   .cui-no-border {
```
Comment (pauljeter) [inline scss/components/content-item.scss]: This should be in the Utilities files, or needs to be specific to the Content Item

Comment (daanishnasir) [reply scss/components/content-item.scss]: Found .ellipsis in utils/core.scss, switching it out now.

Comment (pauljeter) [reply scss/components/content-item.scss]: This comment was referring to the `.cui-no-border` class. It should either be `.content-item--no-border` or a `.no-border` should be added to the utilities.

File: scss/components/content-item.scss (hunk 1 above)
Comment (daanishnasir) [reply scss/components/content-item.scss]: using $brand-shadow variable from settings/core.scss

Comment (bfbiggs) [reply scss/components/content-item.scss]: Change to color variable since that's a legacy variable.

Comment (daanishnasir) [reply scss/components/content-item.scss]: changed to color var, box-shadow: 0 0 0 1px rgba($black, 0.2); @bfbiggs

File: scss/components/content-item.scss (hunk 4 above)
Comment (daanishnasir) [reply scss/components/content-item.scss]: Using mixin called text-overflow that does this. Was already created beforehand.

File: scss/components/content-item.scss [hunk 6]
```diff
@@ -10,17 +12,17 @@
   background-position: center;
   background-repeat: no-repeat;
   background-size: cover;
-  border: 1px solid rgba(0, 0, 0, 0.06);
+  border: 1px solid rgba($black, 0.06);
   border-radius: 4px;
-  box-shadow: inset 1px 0 0 0 rgba(0, 0, 0, 0.06),
-    inset -1px 0 0 0 rgba(0, 0, 0, 0.06);
+  box-shadow: inset 1px 0 0 0 rgba($black, 0.06),
+    inset -1px 0 0 0 rgba($black, 0.06);
```
Comment (bfbiggs) [inline scss/components/content-item.scss]: Look at the colors settings file - line 589.  You'll see the variables you need to replace these rgba values completely.

Comment (daanishnasir) [reply scss/components/content-item.scss]: Replaced

File: scss/components/content-item.scss [hunk 7]
```diff
@@ -1,7 +1,9 @@
 @import '../settings/colors';
 @import '../settings/fonts';
+@import '../settings/core';
```
Comment (daanishnasir) [inline scss/components/content-item.scss]: removing this

File: scss/settings/colors.scss [hunk 8]
```diff
@@ -618,6 +619,7 @@ $white-8: $white-08;
 
 $gray-1-40: rgba($gray-3, 0.4);
 $gray-5-88: rgba($gray-5, 0.88);
+$gray-6-40: rgba($gray-6, 0.4);
```
Comment (pauljeter) [inline scss/settings/colors.scss]: Approving, but not sure how I feel about these variables. (I know that you did not add the 1st two)

**Pull request merged**